src/hotspot/share/gc/z/zCollectedHeap.cpp

```diff
@@ -80,7 +80,7 @@ void ZCollectedHeap::initialize_serviceability() {
 class ZStopConcurrentGCThreadClosure : public ThreadClosure {
 public:
   virtual void do_thread(Thread* thread) {
-    if (thread->is_ConcurrentGC_thread() && !thread->is_Worker_thread()) {
+    if (thread->is_ConcurrentGC_thread()) {
       static_cast<ConcurrentGCThread*>(thread)->stop();
```

Comment:
Should you have a cast function like JavaThread does  ?
  static JavaThread* cast(Thread* t) {
    assert(t->is_Java_thread(), "incorrect cast to JavaThread");
    return static_cast<JavaThread*>(t);
  }

At one point @dholmes replaced all the thread->as_Java_thread() with JavaThread::cast() so this would be consistent with that and nice.

Reply:
Sounds like a good idea. Added `ConcurrentGCThread::cast()`.